core/lib/server/render/pub/renderMarkdownWithPub.ts

```diff
@@ -317,16 +326,12 @@ export const renderMarkdownWithPub = async (
 		.use(remarkRehype)
 		.use(rehypeFormat);
 
-	let processor: Processor<Node, Node, Node, Node, string>;
-
 	if (asMarkdown) {
 		// Convert the HTML back to markdown using rehype-remark
-		processor = processorBase.use(rehypeRemark).use(remarkStringify);
-	} else {
-		processor = processorBase.use(rehypeStringify);
+		const processor = processorBase.use(rehypeRemark).use(remarkStringify);
+		return process(processor, text);
 	}
 
-	const result = await processor.process(text);
-
-	return result.toString().trim();
+	const processor = processorBase.use(rehypeStringify);
+	return process(processor, text);
```

Comment:
this just solved that conditional assignment a bit nicer. This lead to an error, as the fourth `Node` in `Processor` was a more specific thing after using `remarkStringify`, so you couldn't do that.